### PR TITLE
Full data transmission with app

### PR DESCRIPTION
**Pull request by kgliwinski**
ESSA

### CODE REVIEW

File: src/components/controllers/sensor_controller/sensor_controller.c [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+#include <string.h>
+
+#include "adc_wrapper.h"
+#include "macki_log.h"
+#include "sensor_converters.h"
+#include "sensor_driver_definitions.h"
+
+#define TAG "SENSOR_CONTROLLER"
+
+typedef struct {
```
Comment (JSiuda1) [inline src/components/controllers/sensor_controller/sensor_controller.c:14]: you can create static struct here with the sensor_controller_drivers name, and also initialize the values here.
```
static struct {
  ads1115_driver_t* adc_expander;
  lis2dw12_driver_t* accelerometer;
  tmp1075_driver_t* temperature_sensor;
  vl53l0x_driver_t* distance_sensor;
} sensor_controller_drivers = {
    .adc_expander = &adc_expander,
    .accelerometer = &accelerometer,
    .temperature_sensor = &temperature_sensor,
    .distance_sensor = &distance_sensor,
}
```

File: src/components/controllers/sensor_controller/sensor_controller.c [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  rtc_wrapper_init();
+  init_shared_i2c_wrapper();
+
+  bool adc_ret = adc_wrapper_init();
+  if (!adc_ret) {
+    MACKI_LOG_ERROR(TAG, "Failed to initialize ADC wrapper");
+    return false;
+  }
+
+  ring_buffer_status_t rb_ret = ring_buffer_init(&sensor_data_buffer, 256);
```
Comment (JSiuda1) [inline src/components/controllers/sensor_controller/sensor_controller.c:43]: Huge buffer

**Review by JSiuda1 — APPROVED** [review]
G :+1:

File: src/components/controllers/sensor_controller/sensor_controller.c (hunk 2 above)
Comment (kgliwinski) [reply src/components/controllers/sensor_controller/sensor_controller.c:43]: with a hidden feature of resetting one core 💪 💀

**Pull request merged**